build_ngtf.py

```diff
@@ -0,0 +1,400 @@
+    os.chdir(pwd)
+
+
+def install_virtual_env(venv_dir):
+    # Check if we have virtual environment
+    # TODO
+
+    # Setup virtual environment
+    venv_dir = os.path.abspath(venv_dir)
+    # Note: We assume that we are using Python 3 (as this script is also being
+    # executed under Python 3 as marked in line 1)
+    call(["virtualenv", "--system-site-packages", "-p", "python3", venv_dir])
+
+def load_venv(venv_dir):
+    activate_this_file = os.path.abspath(venv_dir) + "/bin/activate_this.py"
+    # The execfile API is for Python 2. We keep here just in case you are on an
+    # obscure system without Python 3
+    # execfile(activate_this_file, dict(__file__=activate_this_file))
+    exec(
+        compile(
+            open(activate_this_file, "rb").read(), activate_this_file, 'exec'),
+        dict(__file__=activate_this_file), dict(__file__=activate_this_file))
+
+def build_tensorflow(venv_dir, src_dir, artifacts_dir):
+
+    install_virtual_env(venv_dir)
+    load_venv(venv_dir)
+
+    # Install the pip packages
+    call([
+        "pip",
+        "install",
+        "-U",
+        "pip",
+        "six",
+        "numpy",
+        "wheel",
+        "mock",
+        "protobuf",
+        "keras_applications==1.0.5",
+        "--no-deps",
+        "keras_preprocessing==1.0.3",
+        "--no-deps",
+    ])
+
+    # Print the current packages
+    call(["pip", "list"])
+
+    base = sys.prefix
+    python_lib_path = os.path.join(base, 'lib', 'python%s' % sys.version[:3],
+                                   'site-packages')
+    python_executable = os.path.join(base, "bin", "python")
+
+    print("PYTHON_BIN_PATH: " + python_executable)
+
+    # In order to build tensotflow, we need to be in the virtual environment
+    pwd = os.getcwd()
+
+    src_dir = os.path.abspath(src_dir)
+    print("Source location: " + src_dir)
+
+    # Update the artifacts directory
+    artifacts_dir = os.path.join(os.path.abspath(artifacts_dir), "tensorflow")
+    print("ARTIFACTS DIR: %s" % artifacts_dir )
+
+    os.chdir(src_dir)
+
+    # Set the TensorFlow configuration related variables
+    os.environ["PYTHON_BIN_PATH"] = python_executable
+    os.environ["PYTHON_LIB_PATH"] = python_lib_path
+    os.environ["TF_NEED_IGNITE"] = "0"
+    os.environ["TF_ENABLE_XLA"] = "1"
+    os.environ["TF_NEED_OPENCL_SYCL"] = "0"
+    os.environ["TF_NEED_COMPUTECPP"] = "0"
+    os.environ["TF_NEED_ROCM"] = "0"
+    os.environ["TF_NEED_MPI"] = "0"
+    os.environ["TF_NEED_CUDA"] = "0"
+    os.environ["TF_DOWNLOAD_CLANG"] = "0"
+    os.environ["TF_SET_ANDROID_WORKSPACE"] = "0"
+    os.environ["CC_OPT_FLAGS"] = "-march=native"
+
+    call([
+        "./configure",
+    ])
+
+    # Build the python package
+    call([
+        "bazel",
+        "build",
+        "--config=opt",
+        #"--config=ngraph",
+        #"--config=noaws",
+        #"--config=nogcp",
+        #"--config=nohdfs",
+        #"--config=noignite",
+        #"--config=nokafka",
```

Comment:
In case the commented out lines indicate the possible options, we could add `#"--config=mkl" as another option.
Also, when the script is able to take more options, with-or-without-mkl could be specified in the argument parser